plugins/modules/cloud/misc/proxmox_snap.py

```diff
@@ -144,7 +176,7 @@ def main():
         vmid=dict(required=False),
         hostname=dict(),
         timeout=dict(type='int', default=30),
-        state=dict(default='present', choices=['present', 'absent']),
+        state=dict(default='present', choices=['present', 'absent', 'rollback']),
```

Comment:
You probably want to add `require_if=[('state', 'rollback', ('snapname', ))],` to the `AnsibleModule()` call below.